moved pmem setting restore before top bar is loaded

gullradriel opened the pull request:
Since top bar is loaded before the rest of the menu, I had to move pmem settings restore just before the top bar is set.
Fix for https://github.com/eried/portapack-mayhem/issues/1099

kallanreed commented on firmware/common/portapack_persistent_memory.cpp:
```diff
@@ -820,6 +820,16 @@ void set_encoder_dial_sensitivity(uint8_t v) {
+    File pmem_flag_file_handle;
+    std::string pmem_flag_file = "/SETTINGS/PMEM_FILEFLAG";
+    auto result = pmem_flag_file_handle.open(pmem_flag_file);
+    if (!result.is_valid()) {
```
is_valid is already a bool. Maybe `return result.is_valid();`

kallanreed replied:
Actually, what is this even doing? It's just returning true if the file opens?

kallanreed replied:
If you just want to verify the file exists, there's a `file_exists` helper I added to file.hpp.

kallanreed replied:
I think the word "save" in the name is confusing me.
Maybe just `should_load_pmem_from_sdcard` or even `has_sdcard_pmem_file` would describe what this is doing better.

gullradriel replied:
I reduced the function to return file_exists( PMEM_FLAGFILE )
Changed the name of the test func to should_use_sdcard_for_pmem

kallanreed commented on firmware/application/ui_navigation.cpp:
```diff
@@ -156,6 +160,9 @@ SystemStatusView::SystemStatusView(
 
     button_back.on_select = [this](ImageButton&) {
+        if (portapack::persistent_memory::save_load_pmem_from_sdcard_flag()) {
+            portapack::persistent_memory::save_persistent_settings_to_file("SETTINGS/pmem_settings");
```
nit: consider a constant for that string. It's used in quite a few places.

gullradriel replied:
Did both use a define for those files and removed the save and load parameter

gullradriel commented:
BTW I also renamed the check_load_mem_at_startup to check_use_sdcard_for_pmem
And changed the text from "load from sd at startup" to "use sdcard for p.mem"
